source/server/common.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/http/header_map.h"
+
+#include "external/envoy/source/common/singleton/const_singleton.h"
+
+#include "api/server/response_options.pb.h"
+
+namespace Nighthawk {
+namespace Server {
+
+namespace TestServer {
+
+class HeaderNameValues {
```

Comment:
Like for above, see 9be53343674899bc8a2e962ebbee192a96ff31f9. However, I kept the Envoy construct, because this is shared/reused across the two extensions, and having the `LowerCaseString` is convenient -- that is the form that consumers are probably going to end up needing when they want access to this.